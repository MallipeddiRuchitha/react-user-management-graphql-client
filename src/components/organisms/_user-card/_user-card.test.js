import React from 'react';
import { create } from "react-test-renderer";
import { render, fireEvent, getByRole } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import UserCard from './_user-card';
import { ApolloProvider } from 'react-apollo'

const client={};

describe("UserCard component",() =>{
    
     
    test("Matches the snapshot",() => {
        const userCard = create(<ApolloProvider client={client}><UserCard /></ApolloProvider>);
        expect(userCard.toJSON()).toMatchSnapshot();

    })
    test("userCard element is rendered",() => {
        const { getByTestId, getByText } = render(<ApolloProvider client={client}><UserCard /></ApolloProvider>);
        expect(getByTestId("userCard")).toBeInTheDocument();
        
    });

    test("DeleteIcon is rendered",() => {
        
        const { getByTestId, getByText,getByRole } = render(<ApolloProvider client={client}><UserCard /></ApolloProvider>);
        expect(getByTestId("deleteIcon")).toBeInTheDocument();
    });
    test("EditIcon is rendered",() => {
        
        const { getByTestId, getByText,getByRole } = render(<ApolloProvider client={client}><UserCard /></ApolloProvider>);
        expect(getByTestId("editIcon")).toBeInTheDocument();
    });
    test(" when EditIcon is clicked input must be rendered",() => {
        
        const { getAllByTestId, getByTestId } = render(<ApolloProvider client={client}><UserCard /></ApolloProvider>);
        fireEvent.click(getByTestId("editIcon"));
        expect(getAllByTestId("input")).toHaveLength(2);
    });

    
});