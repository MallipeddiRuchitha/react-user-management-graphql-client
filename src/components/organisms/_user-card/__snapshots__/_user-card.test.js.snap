// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard component Matches the snapshot 1`] = `
<div>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-styleCard-1 MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="userCard"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    />
    <p
      className="MuiTypography-root makeStyles-email-2 MuiTypography-body1"
    />
    <div>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        data-testid="editIcon"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        data-testid="deleteIcon"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
    </div>
  </div>
</div>
`;
